### PR TITLE
chore(release): Update release testing section for new E2E testing script

### PR DESCRIPTION
**Pull request by kelset**
This PR updates the docs to account for the new RN>=71 local script for E2E testing (this https://github.com/facebook/react-native/pull/34513)

While working on it, I've also cleaned up a few references here and there to things that don't exist anymore (RIP release testers) and overall wordiness :)

### CODE REVIEW

**Comment by kelset** [comment]
should be ready to go @Simek @cortinico

File: website/contributing/release-testing.md [hunk 1]
```diff
@@ -35,100 +48,64 @@ These steps needs to be followed by the release crew as part of the release proc
...
-| Flipper debugger             | ✅/🚨/🙅‍♂️                |
-| Deploy to TestFlight         | ✅/🚨/🙅‍♂️                |
-| **Tested - Android**         |                         |
-| Fast Refresh                 | ✅/🚨/🙅‍♂️                |
-| Debug/dev build on Emulator  | ✅/🚨/🙅‍♂️                |
-| Debug/dev build on Device    | ✅/🚨/🙅‍♂️                |
-| Production build             | ✅/🚨/🙅‍♂️                |
-| Chrome remote debugger       | ✅/🚨/🙅‍♂️                |
-| Hermes debugger              | ✅/🚨/🙅‍♂️                |
-| Flipper debugger             | ✅/🚨/🙅‍♂️                |
```
Comment (cortinico) [inline website/contributing/release-testing.md:133]: I'm a bit afraid on losing this list, which I believe it's still valuable to keep in some form, especially if we onboard external QA in some form

**Review by cortinico — APPROVED** [review]
Change LGTM to me overall 👍

File: website/contributing/release-testing.md (hunk 1 above)
Comment (kelset) [reply website/contributing/release-testing.md:133]: I have it saved in my notes, we can bring it back if needed when the QA ramps up?

Comment (kelset) [reply website/contributing/release-testing.md:133]: the thing is that it's aimed to someone testing RN against a prod app, so I don't know how valuable it'd be to have in the docs here

Comment (cortinico) [reply website/contributing/release-testing.md:133]: > I have it saved in my notes, we can bring it back if needed when the QA ramps up?

I don't have a strong opinion. We can remove it for now and bring it back if needed

Comment (kelset) [reply website/contributing/release-testing.md]: @cortinico readded in its own dedicated section so that we can link it in the "road to" discussion

**Pull request merged**